sympy/printing/pretty/pretty.py

```diff
@@ -1954,8 +1957,9 @@ def _print_Pow(self, power):
             if e is S.NegativeOne:
                 return prettyForm("1")/self._print(b)
             n, d = fraction(e)
-            if n is S.One and d.is_Atom and not e.is_Integer and self._settings['root_notation']:
-                return self._print_nth_root(b, e)
+            if n is S.One and d.is_Atom and not e.is_Integer and not e.is_irrational and self._settings['root_notation']:
```

Comment:
What is the `is_irrational` check for?

I don't think that the assumptions system should be used in the printing code.

Reply:
not `is_irrational` means that symbols, rational numbers, and radicals like `EulerGamma` would be passed to the function. In my current implementation, I am guaranteeing that irrational numbers are printed without the square root, however, all other symbols go through a check, in whether they are simple (height and width 1), in which case they are printed with the square root, otherwise they are also printed as fractions.

What do you mean by the assumptions system?

Reply:
The `is_irrational` attribute is part of the assumptions system:
```python
In [50]: x = Symbol('x', irrational=True)

In [51]: x.is_irrational
Out[51]: True

In [52]: x = Symbol('x', rational=True)

In [53]: x.is_irrational
Out[53]: False
```
The printing code should be based only on structural properties of the expression rather than semantic properties implied by the assumptions system.

Reply:
Can you suggest any way to distinguish between `EulerGamma` and `pi`? I was preferring printing `EulerGamma` over the square root and irrational numbers as fractions, but I am unable to distinguish between `EulerGamma` and `pi` without using `is_irrational`.

Reply:
Why should those print any differently?

Reply:
I think that was the ideology the previous PR author was going for, so I just went along with it. I have made the changes now.